### PR TITLE
test: reproduce crash in #4345

### PR DESCRIPTION
**Pull request by rgrinberg**
@jeremiedimino looks like this is still a problem.

### CODE REVIEW

**Comment by rgrinberg** [comment]
@jeremiedimino any ideas on how to improve the message?

**Comment by jeremiedimino** [comment]
What about adding `human_readable_description` arguments to various function?

For instance, we could imagine the cycle being displayed as:

- evaluating glob file
- analysing directory lib
- calculating install entries from stanzas
- ...
- loading rules for directory .

**Comment by rgrinberg** [comment]
Okay, I added some human readable descriptions. Not sure about how useful the entry for load-dir. It's useful for some errors, but for others it's noise. I suppose it's harmless anyway.

File: test/blackbox-tests/test-cases/reporting-of-cycles.t/run.t [hunk 1]
```diff
@@ -67,3 +67,27 @@ cryptic and can involve unrelated files:
...
+  $ touch lib.opam file lib/lib.ml
+  $ cat >lib/dune <<EOF
+  > (library (name lib) (public_name lib))
+  > (copy_files (files ../file))
+  > EOF
+  $ dune build --root .
+  Error: Dependency cycle between:
+     Loading rules for directory _build/default
+  -> evaluating predicate
+     { dir = In_build_dir "default"
```
Comment (jeremiedimino) [inline test/blackbox-tests/test-cases/reporting-of-cycles.t/run.t:85]: This record is not very user friendly. `human_readable_description` is supposed to print something any user can understand.

File: test/blackbox-tests/test-cases/cinaps/multiple-cinaps.t [hunk 2]
```diff
@@ -24,6 +24,7 @@ Multiple cinaps stanzas in the same dune file
   _build/default/.cinaps.a7811055/cinaps.ml-gen:
   - dune:1
   - dune:2
+  -> required by Loading rules for directory _build/default/.cinaps.a7811055
```
Comment (jeremiedimino) [inline test/blackbox-tests/test-cases/cinaps/multiple-cinaps.t:27]: `Loading` shouldn't be capitalised as it is in the middle of a sentence.

**Review by jeremiedimino — CHANGES_REQUESTED** [review]
Agreed that the `Loading rules ...` is noise most of the time. I don't think we should print it by default.

**Comment by jeremiedimino** [comment]
BTW, there seems to be commits from another PR here. Namely installing directories.

File: test/blackbox-tests/test-cases/reporting-of-cycles.t/run.t (hunk 1 above)
Comment (rgrinberg) [reply test/blackbox-tests/test-cases/reporting-of-cycles.t/run.t]: Changed it to print only the directory.

File: test/blackbox-tests/test-cases/cinaps/multiple-cinaps.t (hunk 2 above)
Comment (rgrinberg) [reply test/blackbox-tests/test-cases/cinaps/multiple-cinaps.t]: removed loading

**Comment by rgrinberg** [comment]
We should really print `load_dir` conditionally, but the current api doesn't allow an optional human readable description. Worth improving in another PR?

**Review by jeremiedimino — APPROVED** [review]
Looks good

**Comment by jeremiedimino** [comment]
> We should really print load_dir conditionally, but the current api doesn't allow an optional human readable description. Worth improving in another PR?

Yes, that seems useful

**Pull request merged**